src/Parsing/Parser.hs

```diff
@@ -189,3 +202,139 @@ someASTParser PHP        = Just (SomeASTParser (ASTParser tree_sitter_php :: Par
+--
+--   This can be used to perform actions on terms supporting some feature abstracted using a typeclass, without knowing (or caring) what the specific term types are.
+data SomeParser c a where
+  SomeParser :: c t => Parser (t a) -> SomeParser c a
+
+goParser' :: c (Term (Sum Go.Syntax)) => (Language, SomeParser c Loc)
+goParser' = (Go, SomeParser goParser)
+
+haskellParser' :: c (Term (Sum Haskell.Syntax)) => (Language, SomeParser c Loc)
+haskellParser' = (Haskell, SomeParser haskellParser)
+
+javascriptParser' :: c (Term (Sum TSX.Syntax)) => (Language, SomeParser c Loc)
+javascriptParser' = (JavaScript, SomeParser tsxParser)
+
+jsonParser' :: c (Term (Sum JSON.Syntax)) => (Language, SomeParser c Loc)
+jsonParser' = (JSON, SomeParser jsonParser)
+
+jsxParser' :: c (Term (Sum TSX.Syntax)) => (Language, SomeParser c Loc)
+jsxParser' = (JSX, SomeParser tsxParser)
+
+markdownParser' :: c (Term (Sum Markdown.Syntax)) => (Language, SomeParser c Loc)
+markdownParser' = (Markdown, SomeParser markdownParser)
+
+phpParser' :: c (Term (Sum PHP.Syntax)) => (Language, SomeParser c Loc)
+phpParser' = (PHP, SomeParser phpParser)
+
+pythonParserALaCarte' :: c (Term (Sum Python.Syntax)) => (Language, SomeParser c Loc)
+pythonParserALaCarte' = (Python, SomeParser pythonParser)
+
+pythonParserPrecise' :: c Py.Term => (Language, SomeParser c Loc)
+pythonParserPrecise' = (Python, SomeParser pythonParserPrecise)
+
+pythonParser' :: (c (Term (Sum Python.Syntax)), c Py.Term) => PerLanguageModes -> (Language, SomeParser c Loc)
+pythonParser' modes = case pythonMode modes of
+  ALaCarte -> (Python, SomeParser pythonParser)
+  Precise  -> (Python, SomeParser pythonParserPrecise)
+
+rubyParser' :: c (Term (Sum Ruby.Syntax)) => (Language, SomeParser c Loc)
+rubyParser' = (Ruby, SomeParser rubyParser)
+
+tsxParser' :: c (Term (Sum TSX.Syntax)) => (Language, SomeParser c Loc)
+tsxParser' = (TSX, SomeParser tsxParser)
+
+typescriptParser' :: c (Term (Sum TypeScript.Syntax)) => (Language, SomeParser c Loc)
+typescriptParser' = (TypeScript, SomeParser typescriptParser)
+
+
+-- | The canonical set of parsers producing à la carte terms.
+aLaCarteParsers
+  :: ( c (Term (Sum Go.Syntax))
+     , c (Term (Sum Haskell.Syntax))
+     , c (Term (Sum JSON.Syntax))
+     , c (Term (Sum Markdown.Syntax))
+     , c (Term (Sum PHP.Syntax))
+     , c (Term (Sum Python.Syntax))
+     , c (Term (Sum Ruby.Syntax))
+     , c (Term (Sum TSX.Syntax))
+     , c (Term (Sum TypeScript.Syntax))
+     )
+  => Map Language (SomeParser c Loc)
+aLaCarteParsers = Map.fromList
+  [ goParser'
+  , haskellParser'
+  , javascriptParser'
+  , jsonParser'
+  , jsxParser'
+  , markdownParser'
+  , phpParser'
+  , pythonParserALaCarte'
+  , rubyParser'
+  , typescriptParser'
+  , tsxParser'
+  ]
+
+-- | The canonical set of parsers producing precise terms.
+preciseParsers :: c Py.Term => Map Language (SomeParser c Loc)
+preciseParsers = Map.fromList
```

Comment:
Nit: you could turn on `OverloadedLists`, if you wanted.

Reply:
I do _not_ want 😆